core/src/main/java/ai/timefold/solver/core/impl/domain/variable/declarative/AffectedEntitiesUpdater.java

```diff
@@ -122,21 +89,19 @@ private boolean updateShadowVariable(EntityVariablePair<Solution_> entityVariabl
         if (loopDescriptor != null) {
             var oldLooped = (boolean) loopDescriptor.getValue(entity);
             if (oldLooped != isLooped) {
-                // Loop status change; add to affected entities
-                affectedEntities.add(entityVariable);
```

Comment:
This was indirection. Put into a set, affect later. Seemingly unnecessary, since the invariant is that the loop status changed from previous, and therefore we need to execute the update even if we had already executed it on this entity before. (The set wouldn't save any work.)